mesop/server/state_session.py

```diff
@@ -89,15 +93,37 @@ def save(self, token: str, states: States):
   def clear_stale_sessions(self):
     stale_keys = set()
 
+    # Avoid trying to clear sessions with every request to avoid unnecessary processing.
     current_time = _current_datetime()
-    for key, (timestamp, _) in self.cache.items():
+    if (
+      self.last_checked_sessions + timedelta(minutes=self._SESSION_TTL_MINUTES)
+      > current_time
+    ):
+      return
+
+    self.last_checked_sessions = current_time
+
+    # Add a mutex in the case where a cache item may be removed while looping through
```

Comment:
it's not clear to me how beneficial this mutex is - it seems like either it's too short (you should take a mutex out for the entire block (e.g. up until you delete the cache key in L122) or you should just skip having a mutex and accept a race because you're doing error-handling in L123. WDYT?

Reply:
Hmm, yeah that's a good point. Since I'm not doing a loop anymore, probably wouldn't run into the data changed while looping runtime error now. I'll remove it.